src/error/index.ts

```diff
@@ -0,0 +1,32 @@
+export enum ErrorType {
+  UnknownError = "UnknownError",
+  ValidationError = "ValidationError",
+  InitError = "InitError"
+}
+
+export type ErrorConstructor = {
+  message: string;
+  errorType: ErrorType | string;
+  error?: any;
+  logErrors?: boolean;
+};
+
+export class SquidError extends Error {
+  errorType: string;
+  error: any;
+  logErrors = false;
```

Comment:
Please use standard logLevel notation
info, error, debug and configuration option should be named `logLevel="level"` and `logging=true/false`